src/context/categories/categories-context.js

```diff
@@ -9,7 +9,14 @@ const CategoriesContextProvider = ({ children }) => {
   const [categories, setCategories] = useState([]);
 
   const { loading, error } = useQuery(getAllCategoriesQuery, {
-    onCompleted: (data) => setCategories(data.getAllCategories.items)
+    onCompleted: (data) => {
+      const categoriesList = data.getAllCategories.items;
+      categoriesList.forEach((categorie) => {
```

Comment:
It is not a good idea to mutate an argument incoming in function, in our situation it is a category. Better use the map and apply a new array for setCategories.

Reply:
Got it. Refactored.